source/common/upstream/health_checker_base_impl.cc

```diff
@@ -274,6 +275,9 @@ void HealthCheckerImplBase::ActiveHealthCheckSession::handleSuccess(bool degrade
   num_unhealthy_ = 0;
 
   HealthTransition changed_state = HealthTransition::Unchanged;
+  if (first_check_ || host_->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC)) {
+    host_->setLastHcPassTime(time_source_.monotonicTime());
+  }
```

Comment:
I'm wondering if we should actually do this in the block below when we reach the healthy threshold? It seems like we should start slow start when the host actually becomes healthy? wdyt?

Reply:
yes, did not realise that hosts have `Host::HealthFlag::FAILED_ACTIVE_HC` flag set during initialisation